src/core/params.ts

```diff
@@ -0,0 +1,50 @@
+import { assertUnreachable } from "./assert"
+import * as t from "./types"
+
+export function collectParams(root: t.Num): {
```

Comment:
might be simpler just to return a `Set<t.Param>` or a `t.Param[]` and let the caller filter on `param.fixed`?